include/HAL/TargetSystem.h

```diff
@@ -139,6 +141,31 @@ class Target {
   /// @brief Disable(stop) ongoing timers
   void disableTiming();
 
+  // Diagnostic access
+  /// @brief Add a diagnostic to this target, not thread safe
+  void addDiagnostic(const qssc::Diagnostic &diag) {
+    diagnostics_.emplace_back(diag);
+  }
+  /// @brief Add a diagnostic to this target using mutex locks for thread safety
+  void addDiagnosticWithLock(const qssc::Diagnostic &diag) {
+    const std::lock_guard<std::mutex> lock(diagnosticsMutex_);
+    diagnostics_.emplace_back(diag);
+  }
+  /// @brief Get the diagnostics that this target holds
```

Comment:
Perhaps we should only have this version?

Reply:
I think what I'm proposing is just a bit more forwarding in the manager to accumulate the errors in the target manager.

Reply:
I moved diagnostic emission to the target compilation manager.